drivers/volume/portworx/portworx.go

```diff
@@ -480,6 +479,42 @@ func (d *portworx) WaitForNode(n node.Node) error {
 			}
 		}
 
+		logrus.Infof("px on node %s is now up. status: %v", pxNode.Id, pxNode.Status)
+
+		return "", false, nil
+	}
+
+	if _, err := task.DoRetryWithTimeout(t, 2*time.Minute, 10*time.Second); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (d *portworx) WaitDriverDownOnNode(n node.Node) error {
+	t := func() (interface{}, bool, error) {
+		pxNode, err := d.getClusterManager().Inspect(n.VolDriverNodeID)
+		if err != nil {
+			if regexp.MustCompile(`.+timeout|connection refused.*`).MatchString(err.Error()) {
```

Comment:
Could this be a false alarm? If the service was hooked onto node n1 (down) and we are checking for n2 (up),  then we will report false status.

Reply:
fixed in incremental